### DISCUSSION TITLE
Using query parameters partition in by/count endpoints in Federation API

### DISCUSSION THREAD

**Discussion by asafieva** [CBIIT/ccdi-federation-api#96] [API Design Discussion]
A query parameter "partition" has been introduced in "/by/<>/count" API endpoints. The response structure from the same endpoint now depends on if there were a query parameter **_partition_** or not in the request URL. 

We have now only one Federation node with multiple namespaces in V1, just to mention.

Spec:
```
 responses.by.count.sample.Response:
 oneOf:
 - $ref: '#/components/schemas/responses.by.count.sample.Results'
 - $ref: '#/components/schemas/responses.by.count.sample.NamespacePartitionedResults'
```
A JSON defined by _responses.by.count.sample.Results_ would be returned if "partition" URL Query parameter is not used, and _responses.by.count.sample.NamespacePartitionedResults_ defined JSON (an array) would be returned if the URL Query parameter "partition" was used.

If there is no plan by Federation nodes to implement this "partition" query parameter, **can it be taken out from V1** spec and considered to be included with a better design in V2?

Say, would it be simpler to introduce separate endpoints with parameter "partition" being a URL path parameter, as a separate API endpoint?

 /by/{field}/count >> responses.by.count.sample.Results - an object
 /by/{field}/count/{partition} >> responses.by.count.sample.NamespacePartitionedResults - an array

**Comment by claymcleod** [comment]
This is an artifact from a solution I implemented stemming from me not understanding a requested feature correctly. I ended up including it anyways because (a) others had already reviewed it and it was somewhat nice to have and (b) I had already done the work for it. That being said, unless anyone has a strong opinion that it should be kept in, I don't mind removing it.
